cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultScheduleBuilder.java

```diff
@@ -118,4 +118,11 @@ public ScheduleBuilder waitUntilMet() {
     constraints.get(constraints.size() - 1).setWaitUntilMet(true);
     return this;
   }
+
+  @Override
+  public ScheduleBuilder abortIfNotMet() {
+    // user will only be able to call abortIfNotMet right after they add a Constraint
+    constraints.get(constraints.size() - 1).setWaitUntilMet(false);
+    return this;
```

Comment:
`waitUntilMet` and `abortIfNotMet` can only be called right after a method that adds a constraint. Other builder methods have return type that lack these two methods.